build.sbt

```diff
@@ -99,18 +99,61 @@ lazy val publishSettings = Seq(
   )
 )
 
-lazy val allSettings = buildSettings ++ baseSettings ++ publishSettings ++ pgpSettings
+val scalazVersion = "7.1.4"
+val specs2Version = "3.6.4"
+val scalacheckVersion = "1.12.5"
+
+lazy val allSettings =
+  buildSettings ++ baseSettings ++ publishSettings ++ pgpSettings
+
+lazy val noPublishSettings = Seq(
+  publish := (),
+  publishLocal := (),
+  publishArtifact := false
+)
 
 lazy val root = (project in file("."))
+  .aggregate(core, scalacheck, tests)
+  .settings(noPublishSettings)
```

Comment:
so if you run `publish` in root, what happens, do all the sub-projects get published or does nothing get published?